engine/new_runtime/src/types.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(clippy::missing_safety_doc)]
+
+mod bytes;
+mod const_eq;
+mod rand;
+mod zero;
+
+pub use bytes::{Bytes, ContiguousBytes};
+pub use const_eq::ConstEq;
+pub use rand::Randomized;
+pub use zero::Zeroed;
+
+/// Implements the traits [`Bytes`] onto primitive types and slices.
+macro_rules! impls {
+    ($($type:ty),* ; $size:tt) => {$(
+        impls!{prim  $type}
+        impls!{array $type; $size}
+    )*};
+
+    (prim $type:ty) => {
+        unsafe impl Bytes for $type {}
+    };
+
+    (array $type:ty; ($($size:tt)*)) => {$(
+        #[allow(trivial_casts)]
+        unsafe impl Bytes for [$type; $size] {}
```

Comment:
I think you can use [const generics](https://blog.rust-lang.org/inside-rust/2021/09/06/Splitting-const-generics.html) here

Reply:
Had the same thought myself when I was reviewing the code the other day and remembered that they had added the const  generics.  If I might rewrite this part upstream and commit it back to stronghold.